extension/chrome/elements/compose-modules/compose-my-pubkey-module.ts

```diff
@@ -44,18 +46,40 @@ export class ComposeMyPubkeyModule extends ViewModule<ComposeView> {
       return;
     }
     (async () => {
-      const contacts = await ContactStore.get(undefined, this.view.recipientsModule.getRecipients().map(r => r.email));
-      for (const contact of contacts) {
-        if (contact?.has_pgp && contact.client !== 'cryptup') {
+      const senderEmail = this.view.senderModule.getSender();
+      const senderKi = await this.view.storageModule.getKey(senderEmail);
+      // if we have cashed this longid in this.wkdLongids, setAttachPreference(false) rightaway and return
+      const cached = this.wkdLongids[senderEmail];
+      if (Array.isArray(cached) && cached.some(id => id === senderKi.longid)) {
+        this.setAttachPreference(false);
+        return;
+      }
+      const myDomain = Str.getDomainFromEmailAddress(senderEmail);
+      const foreignRecipients = this.view.recipientsModule.getRecipients().map(r => r.email)
+        .filter(Boolean)
+        .filter(email => myDomain !== Str.getDomainFromEmailAddress(email));
+      if (foreignRecipients.length > 0) {
+        if (!Array.isArray(cached)) {
+          // slow operation -- test WKD for our own key and cache the result
+          const { keys } = await this.view.pubLookup.wkd.rawLookupEmail(senderEmail);
+          const longids = keys.map(key => OpenPGPKey.fingerprintToLongid(key.id));
+          this.wkdLongids[senderEmail] = longids;
+          // check fingerprint
+          if (longids.some(id => id === senderKi.longid)) {
+            this.setAttachPreference(false);
+            return;
+          }
+        }
+        for (const recipient of foreignRecipients) {
           // new message, and my key is not uploaded where the recipient would look for it
-          if (! await this.view.recipientsModule.doesRecipientHaveMyPubkey(contact.email)) {
+          if (! await this.view.recipientsModule.doesRecipientHaveMyPubkey(recipient)) {
             // either don't know if they need pubkey (can_read_emails false), or they do need pubkey
```

Comment:
This is outdated comment (already before this PR), can just say 

_// they do need pubkey_